### PR TITLE
Properly parse alignment shorthands (fixes #16391)

### LINKED ISSUE #16391: alignment shorthand parsing is incorrect
When https://github.com/servo/servo/pull/16322 landed we fixed some Gecko failures, but picked up a bunch of new ones:

> FAIL | layout/style/test/test_property_syntax_errors.html | invalid value 'center safe' not accepted for 'place-content' property
got "center safe", expected ""

> FAIL | layout/style/test/test_property_syntax_errors.html | invalid value 'unsafe start' not accepted for 'place-content' property
got "start unsafe", expected "" 

@tamamu @upsuper

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16558)

### CODE REVIEW

**Comment by emilio** [comment]
This looks good to me. Try run (along with the linear-gradient stuff) is here, feel free to r=me if it's green (well, if it only has unexpected passes), and I'll update expectations right-after.

**Comment by emilio** [comment]
Err, here: https://treeherder.mozilla.org/#/jobs?repo=try&revision=0505bdf71c7c5d44e084e2764813da97c7b362fa

**Review by emilio — COMMENTED** [review]
r=me with that nit (and similar ones around)

File: components/style/values/specified/align.rs [hunk 1]
```diff
@@ -155,6 +155,13 @@ impl AlignJustifyContent {
         AlignFlags::from_bits((self.0 >> ALIGN_ALL_SHIFT) as u8)
             .expect("AlignJustifyContent must contain valid flags")
     }
+
+    /// Whether this value has extra flags.
+    #[inline]
+    pub fn has_extra_flags(self) -> bool {
+        self.primary().bits & ALIGN_FLAG_BITS.bits != 0 ||
```
Comment (nox) [inline components/style/values/specified/align.rs]: Oh. Neat.

**Pull request merged**